shell/common/rasterizer.cc

```diff
@@ -189,6 +189,9 @@ RasterStatus Rasterizer::Draw(
       };
 
   PipelineConsumeResult consume_result = pipeline->Consume(consumer);
+  if (consume_result == PipelineConsumeResult::NoneAvailable) {
+    return RasterStatus::kFailed;
```

Comment:
Is kFailed the right status here? Should we be adding another enumeration value like `kEmpty` or something?

/cc @blasten @iskakaushik @chinmaygarde @cyanglaz for input as well

Reply:
This return value is not actually used, it was introduced in this PR and is currently only used for testing https://github.com/flutter/engine/pull/28383. I tend to use kFailed because the empty layer tree is an abnormal branch and we shouldn't let it appear in the future.

Reply:
Based on additional discussion I'm good with this.

We should take additional steps to avoid this condition ever happening - we should not insert an empty layer tree into the pipeline, and treating this as a failure case makes sense now.